nucypher/network/server.py

```diff
@@ -222,13 +225,40 @@ def set_policy(self, id_as_hex, request: Request):
         # TODO: Sign the arrangement here.  #495
         return  # TODO: Return A 200, with whatever policy metadata.
 
+    def revoke_arrangement(self, id_as_hex, request: Request):
+        """
+        REST endpoint for revoking/deleting a KFrag from a node.
+        TODO: Have Ursula return a signed receipt.
+        """
+        from nucypher.policy.models import RevocationNotice
+
+        revocation_notice = RevocationNotice.from_bytes(request.body)
+
+        try:
+            with ThreadedSession(self.db_engine) as session:
+                # Verify the Notice was signed by Alice
+                policy_arrangement = self.datastore.get_policy_arrangement(
+                    revocation_notice.arrangement_id.hex().encode(),
+                    session=session)
+                alice_pubkey = UmbralPublicKey.from_bytes(
+                    policy_arrangement.alice_pubkey_sig.key_data)
+                is_valid = revocation_notice.verify(alice_pubkey)
+
+                if is_valid:
+                    self.datastore.del_policy_arrangement(
+                        revocation_notice.arrangement_id.hex().encode(),
+                        session=session)
+        except:
```

Comment:
I think, we should specify the exceptions so that we don't eat up any unexpected ones

Reply:
:+1: